src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayPersistence.cs

```diff
@@ -15,6 +18,11 @@ internal InMemoryGatewayPersistence()
         /// </summary>
         protected internal override void Setup(FeatureConfigurationContext context)
         {
+            if (!PersistenceStartup.HasSupportFor<StorageType.GatewayDeduplication>(context.Settings))
```

Comment:
I talked with @danielmarbach and explained the point I was trying to make here: The code that checks for gateway storage belongs in the Gateway feature itself, not in the InMemory persistence implementation. The check and the test should be moved to the GW.